### PR TITLE
Route features

### PR DESCRIPTION
**Pull request by jeabakker**

### CODE REVIEW

File: engine/lib/pagehandler.php [hunk 1]
```diff
@@ -51,6 +51,34 @@ function elgg_get_route(string $name): ?\Elgg\Router\Route {
...
+		$route_info = _elgg_services()->urlMatcher->match($path);
+		if (!isset($route_info['_route'])) {
+			return null;
+		}
+		
+		$route = _elgg_services()->routes->get($route_info['_route']);
+		$route->setMatchedParameters($route_info);
+		
+		return $route;
+	} catch (\Symfony\Component\Routing\Exception\ExceptionInterface $e) {
```
Comment (jdalsem) [inline engine/lib/pagehandler.php:76]: use throwable... we always want `null`

File: mod/discussions/classes/Elgg/Discussions/Menus/Filter.php [hunk 2]
```diff
@@ -25,12 +25,12 @@ public static function filterTabsForDiscussions(\Elgg\Hook $hook): ?MenuItems {
 		$entity_type = $hook->getParam('entity_type', '');
 		$entity_subtype = $hook->getParam('entity_subtype', '');
 		if (empty($entity_type) || empty($entity_subtype)) {
-			$route = elgg_get_current_route();
-			if ($route instanceof Route) {
```
Comment (jdalsem) [inline mod/discussions/classes/Elgg/Discussions/Menus/Filter.php:29]: no use statement to clean up?

File: mod/friends/classes/Elgg/Friends/Menus/Filter.php [hunk 3]
```diff
@@ -34,12 +34,12 @@ public static function registerFilterTabs(\Elgg\Hook $hook): ?MenuItems {
 		$entity_type = $hook->getParam('entity_type', '');
 		$entity_subtype = $hook->getParam('entity_subtype', '');
 		if (empty($entity_type) || empty($entity_subtype)) {
-			$route = elgg_get_current_route();
-			if ($route instanceof Route) {
```
Comment (jdalsem) [inline mod/friends/classes/Elgg/Friends/Menus/Filter.php:38]: no use statement to cleanup?

File: engine/classes/Elgg/Menus/Filter.php [hunk 4]
```diff
@@ -101,12 +101,12 @@ public static function registerFilterTabs(\Elgg\Hook $hook): MenuItems {
 		$entity_type = $hook->getParam('entity_type', '');
 		$entity_subtype = $hook->getParam('entity_subtype', '');
 		if (empty($entity_type) || empty($entity_subtype)) {
-			$route = elgg_get_current_route();
-			if ($route instanceof Route) {
```
Comment (jdalsem) [inline engine/classes/Elgg/Menus/Filter.php:105]: no use statement to cleanup?

**Review by jdalsem — CHANGES_REQUESTED** [review]

File: engine/lib/pagehandler.php (hunk 1 above)
Comment (jeabakker) [reply engine/lib/pagehandler.php:76]: catching only Routing exceptions, the rest will keep on throwing

**Review by jdalsem — APPROVED** [review]

**Pull request merged**